src/orquestra/sdk/_ray/_client.py

```diff
@@ -112,6 +113,21 @@ def run_dag_async(
         ):
             ray.workflow.run_async(dag_node, workflow_id=workflow_id, metadata=metadata)
 
+            # Give some time to ray to actually submit workflow
+            submission_timeout_in_sec = 10
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏Does this mean submitting to Ray could take up to 10 seconds?

Reply:
It always could. This is only the timer that says "if it takes LONGER than 10 seconds, error out". This wont extent the length of submission itself - only the length of command so it waits until workflow is actually submitted.